src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,28 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+
+  const onChangeStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterActions.setStatus(event.target.value as Status));
```

Comment:
Direct casting of event.target.value as Status can be unsafe if the value does not match the Status enum. It would be better to validate the value before dispatching the action.